packages/core/src/geometry/BufferSystem.ts

```diff
@@ -0,0 +1,173 @@
+import { System } from '../System';
+import { GLBuffer } from './GLBuffer';
+
+import type { Renderer } from '../Renderer';
+import type { IRenderingContext } from '../IRenderingContext';
+import type { Buffer } from './Buffer';
+
+/**
+ * System plugin to the renderer to manage buffers.
+ *
+ * WebGL uses Buffers as a way to store objects to the GPU.
+ * This system makes working with them a lot easier.
+ *
+ * Buffers are used in two main places in WebGL
+ * - geometry information
+ * - Uniform information (via uniform buffer objects - a webGL 2 only feature)
+ * - Transform feedback information.
+ *
+ * This system will handle the binding of buffers to the GPU as well as uploading
+ * them. With this system, you never need to work directly with GPU buffers, but instead work with
+ * the PIXI.Buffer class.
+ *
+ *
+ * @class
+ * @extends PIXI.System
+ * @memberof PIXI
+ */
+export class BufferSystem extends System
+{
+    CONTEXT_UID: number;
+    gl: IRenderingContext;
+
+    readonly managedBuffers: {[key: number]: Buffer};
+
+    /**
+     * @param {PIXI.Renderer} renderer - The renderer this System works for.
+     */
+    constructor(renderer: Renderer)
+    {
+        super(renderer);
+
+        /**
+         * Cache for all buffers by id, used in case renderer gets destroyed or for profiling
+         * @member {object}
+         * @readonly
+         */
+        this.managedBuffers = {};
+    }
+
+    /**
+     * Sets up the renderer context and necessary buffers.
+     */
+    protected contextChange(): void
+    {
+        this.disposeAll(true);
+
+        this.gl = this.renderer.gl;
+
+        // TODO fill out...
+        this.CONTEXT_UID = this.renderer.CONTEXT_UID;
+    }
+
+    /**
+     * this binds specified buffer. OOn first run, it will create the webGL buffers for the context too
+     *
+     * @param {PIXI.Buffer} buffer the buffer to bind to the renderer
+     */
+    bind(buffer:Buffer):void
```

Comment:
I'm noticing you don't leave any space between the identifier and type annotation. Let's give it some breathing room, just for consistency mate 🍺 (this isn't the only place)

```suggestion
    bind(buffer: Buffer): void
```

Reply:
hmm.. feels like we could use lint auto fix for these ones! Does it fix these for you automatically? (just to understand if its my set up, or the lint rules?)

Reply:
lint rule updated!